sql/catalyst/src/test/scala/org/apache/spark/sql/catalyst/optimizer/CombiningLimitsSuite.scala

```diff
@@ -117,4 +128,90 @@ class CombiningLimitsSuite extends PlanTest {
       testRelation.select().groupBy()(count(1)).orderBy(count(1).asc).analyze)
     comparePlans(optimized4, expected4)
   }
+
+  test("SPARK-33497: Override maxRows in some LogicalPlan") {
```

Comment:
Could you split this test into pieces then add clearer test title for them?

Reply:
Also, could you add tests to check if an expected max row value is propagated correctly?

Reply:
Yes, it will better.